src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateClone = Object.assign({}, state);
+
+  for (const act of actions) {
+    switch (act.type) {
+      case 'addProperties':
+        Object.assign(stateClone, act.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of act.keysToRemove) {
+          delete stateClone[key];
+        }
+        break;
+
+      case 'clear':
+        stateClone = {};
+        break;
+    }
```

Comment:
According to the checklist, a default case should be added in the switch statement for error handling. This ensures that the function can properly handle unexpected values for the action type.